### PR TITLE
core: frontend: Improve main menu for small and big screens 

### PR DESCRIPTION
**Pull request by patrickelectric**
Peek 28-02-2022 18-08

### CODE REVIEW

File: core/frontend/src/App.vue [hunk 1]
```diff
@@ -41,97 +36,116 @@
...
+      class="fill-height"
+      no-gutters
     >
-      <v-container
-        elevation="0"
-        class="d-flex justify-center"
+      <v-navigation-drawer
+        id="navigator-drawer"
+        permanent
+        :expand-on-hover="is_small_screen"
```
Comment (Williangalvani) [inline core/frontend/src/App.vue:46]:
```suggestion
        :expand-on-hover="$vuetify.breakpoint.mdAndDown"
```

Comment (Williangalvani) [reply core/frontend/src/App.vue:46]: we shold use breakpoints where we can:
https://vuetifyjs.com/en/features/breakpoints/#thresholds

**Review by Williangalvani — CHANGES_REQUESTED** [review]

File: core/frontend/src/App.vue (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/App.vue:46]: Good catch

**Comment by patrickelectric** [comment]
There is still a problem where when the settings/power/feedback options are open while the screen is small, on the button is clicked, the navbar lost focus, goes to the minimized size and the dialog that just opened get closed.

**Comment by rafaellehmkuhl** [comment]
I would like to use the mini-variant collapsed for medium screens, and hiding the drawer only for smaller ones, but the problem in this case is that all the power-like buttons and the git info have to be hidden when the drawer is not hovered, and the mini-variant API apparently has a bug where the sync property, which allows us to listen for hovering, is not two-way-binded, and when used we cannot force the mini-variant just for some specific screen size. I'm making a minimum-code example to reproduce this bug and try to get some help from the vuetify team.

I ended up not using the mini-variant and just collapsing the drawer when the screen is medium-down and making it fixed on bigger screens. This behavior leaves more space to the main content.

Behavior was tested and adapted to make the tour work without problems.

ezgif-2-b47c9e41b8

File: core/frontend/src/App.vue [hunk 2]
```diff
@@ -211,7 +212,7 @@ export default Vue.extend({
 
   data: () => ({
     settings,
-    drawer: false,
+    drawer: undefined as boolean|undefined,
```
Comment (patrickelectric) [inline core/frontend/src/App.vue:214]: What is the reason to `drawer` to be an union and not a simple boolean ?

Comment (rafaellehmkuhl) [reply core/frontend/src/App.vue:214]: With this patch, the drawer can start as both open or close, depending on the user screen size. As we don't know how it is at start, we make it initial value undefined, and the v-model will take care of updating at mounting time.

**Pull request merged**